torch/_inductor/codegen/cpp.py

```diff
@@ -1481,7 +1491,15 @@ def index_expr(expr, dtype):
         if stride == 0:
             return CppOverrides.index_expr(expr, dtype)
         elif stride is not None:
-            value = ops.to_dtype(cexpr(index), dtype)
+            bounds = ValueRanges.unknown()
+            # If this expression does not come from an FX node, we compute its bounds
+            if (
+                fx_node := getattr(V.interpreter, "current_node", None)
+            ) and fx_node.target != "index_expr":
+                bounds = bound_sympy(index)
+
+            idx = V.kernel.cse.generate(V.kernel.compute, cexpr(index), bounds=bounds)
```

Comment:
nit: perhaps we can factor out a util function for this, same code patterns in several places.